### PR TITLE
Cleanup PR

### PR DESCRIPTION
**Pull request by pomadchin**
Cleanup build.sbt // "provided flags"
Added test

### CODE REVIEW

File: build.sbt
Comment (metasim) [inline build.sbt]: Before I merge this one, I need to understand the way you guys use `provided`, as I've only seen it used in the context of App Containers and other runtime frameworks, never in a situation where just a library is involved. What is the benefit, and how do you "provide" those libraries for an executable? 

Edit: 
I can understand having Spark as `provided`, but not `spark-avro`, as upstream users of this library may not know (or want to know) that it should be manually added to the runtime classpath.

CC @mteldridge

File: src/test/scala/astraea/spark/avro/AvroDerivedSparkEncoderSpec.scala
Comment (metasim) [inline src/test/scala/astraea/spark/avro/AvroDerivedSparkEncoderSpec.scala:301]: Great idea for catching those cases. 👍

File: src/test/scala/astraea/spark/TestEnvironment.scala
Comment (metasim) [inline src/test/scala/astraea/spark/TestEnvironment.scala:44]: Thanks for the migration.

File: build.sbt
Comment (metasim) [reply build.sbt]: PS: Here's what the Maven docs have to say about `provided`:

> This is much like compile, but indicates you expect the JDK or a container to provide the dependency at runtime. For example, when building a web application for the Java Enterprise Edition, you would set the dependency on the Servlet API and related Java EE APIs to scope provided because the web container provides those classes. This scope is only available on the compilation and test classpath, and is not transitive.

In my opinion that general purpose libraries should almost *never* use `provided`, but leave that to the application level, as the determination of what will be available at runtime is an application-level design decision. It is my understanding that if an application-level build declares a downstream (transitive) dependency as `provided`, then that overrides downstream scope specifiers.

Comment (metasim) [reply build.sbt]: Confirmed that Ivy `provided` ~~scope~~ configuration is interpreted the same:

http://stackoverflow.com/a/7116577/296509

**Review by metasim — COMMENTED** [review]
If you drop the "provided" scope specifiers I can merge this in.

File: build.sbt
Comment (pomadchin) [reply build.sbt]: I agree with you about `spark-avro` dep; but my intention was to exclude geotrellis from this library as from a client ready library (beta / alpha client ready); as geotrellis is still a snapshot version, don't think it should strongly depend on gt version at least for now. So there would easier to choose preferred gt, spark and hadoop; and no needs to exclude it in order to put everything into fat assembly (in case of spar-submit usage)

**Review by metasim — APPROVED** [review]

**Pull request merged**
